pyprep/utils.py

```diff
@@ -20,6 +21,71 @@ def _intersect(list1, list2):
     return list(set(list1).intersection(set(list2)))
 
 
+def _mat_quantile(arr, q, axis=None):
+    """Calculate the numeric value at quantile (q) for a given distribution.
+
+    Parameters
+    ----------
+    arr : np.ndarray
+        Input array containing samples from the distribution to summarize.
+    q : float
+        The quantile to calculate for the input data. Must be between 0 and 1,
+        inclusive.
+    axis : {int, tuple of int, None}, optional
+        Axis along which quantile values should be calculated. Defaults to
+        calculating the value at the given quantile for the entire array.
+
+    Returns
+    -------
+    quantile : scalar or np.ndarray
+        If no axis is specified, returns the value at quantile (q) for the full
+        input array as a single numeric value. Otherwise, returns an
+        ``np.ndarray`` containing the values at quantile (q) for each row along
+        the specified axis.
+
+    Notes
+    -----
+    MATLAB calculates quantiles using different logic than Numpy: Numpy treats
+    the provided values as a whole population, whereas MATLAB treats them as a
+    sample from a population of unknown size and adjusts quantiles accordingly.
+    This function mimics MATLAB's logic to produce identical results.
+
+    """
+    q = np.asarray(q, dtype=np.float64)
+    n = len(arr)
+    q_adj = ((q - 0.5) * n / (n - 1)) + 0.5
+    return np.quantile(arr, np.clip(q_adj, 0, 1), axis=axis)
+
+
+def _mat_iqr(arr, axis=None):
+    """Calculate the inter-quartile range (IQR) for a given distribution.
+
+    Parameters
+    ----------
+    arr : np.ndarray
+        Input array containing samples from the distribution to summarize.
+    axis : {int, tuple of int, None}, optional
```

Comment:
where did you get the idea to document the different options in curly brackets?

I usually do: `axis : int | tuple of int | None`, and then write what it defaults to in the description string below (not writing "optional" explicitly)

which doesn't mean that that's better (hence my question).

Reply:
Oh, I just copy/pasted that bit directly from the [docstring of `numpy.quantile`](https://github.com/numpy/numpy/blob/v1.20.0/numpy/lib/function_base.py#L3827-L3931) since that arguments getting passed unmodified to that function anyway. No rationale whatsoever on my part, just figured I'd go with whatever the Numpy maintainers thought was the correct format (I'm new to Numpy docstrings, I've previously only used Google-style ones for Python).

If you'd like I can change it to you preferred format for the sake of consistency across PyPREP!

Reply:
ah thanks, I read the [numpydoc styleguide](https://numpydoc.readthedocs.io/en/latest/format.html) again, and it seems like the curly brackets is advertised. I think I picked up the `|` style from MNE-Python. Let's stick with the curly braces here and in general slowly convert to that style whenever we touch (git diff) a line that still uses the `|` style in the remaining codebase.